datafusion/core/src/datasource/physical_plan/mod.rs

```diff
@@ -63,14 +64,53 @@ use crate::{
 };
 
 use arrow::datatypes::{DataType, SchemaRef};
+use datafusion_common_runtime::SpawnedTask;
+use datafusion_execution::{SendableRecordBatchStream, TaskContext};
 use datafusion_physical_expr::expressions::Column;
 use datafusion_physical_expr::PhysicalSortExpr;
 use datafusion_physical_expr_common::sort_expr::LexOrdering;
+use datafusion_physical_plan::insert::DataSink;
 
+use async_trait::async_trait;
 use futures::StreamExt;
 use log::debug;
 use object_store::{path::Path, GetOptions, GetRange, ObjectMeta, ObjectStore};
 
+/// General behaviors for files that do `DataSink` operations
+#[async_trait]
+pub trait FileSink: DataSink {
+    /// Retrieves the file sink configuration.
+    fn config(&self) -> &FileSinkConfig;
+
+    /// Spawns writer tasks and uses `tokio::join` to collect results.
```

Comment:
It might be helpful to document what a Demux task is here on this trait

Reply:
Agreed, let's do this @mertak-synnada